safe/gis/vector/prepare_vector_layer.py

```diff
@@ -166,7 +166,11 @@ def _check_value_mapping(layer, exposure_key=None):
         classification = layer.keywords['classification']
 
     exposure_classification = definition(classification)
-    other = exposure_classification['classes'][-1]['key']
+
+    if exposure_classification['key'] == 'data_driven_classes':
+        other = 'other'
```

Comment:
AFAIK, this is not necessary as you have your own data driven classes. The other group shouldn't be used, as you are mapping each one to its own class.